src/core/inlines.js

```diff
@@ -129,35 +127,30 @@ export function run(conf) {
     if (subtxt.length === 1) continue;
 
     const df = document.createDocumentFragment();
-    while (subtxt.length) {
-      const t = subtxt.shift();
-      let matched = null;
-      if (subtxt.length) matched = subtxt.shift();
-      df.appendChild(document.createTextNode(t));
-      if (matched) {
-        if (
-          /MUST(?:\s+NOT)?|SHOULD(?:\s+NOT)?|SHALL(?:\s+NOT)?|MAY|(?:NOT\s+)?REQUIRED|(?:NOT\s+)?RECOMMENDED|OPTIONAL/.test(
-            matched
-          )
-        ) {
-          const node = inlineRFC2119Matches(matched);
-          df.appendChild(node);
-        } else if (matched.startsWith("{{{")) {
-          const node = inlineXrefMatches(matched);
-          df.appendChild(node);
-        } else if (matched.startsWith("[[")) {
-          const nodes = inlineBibrefMatches(matched, txt, conf);
-          df.append(...nodes);
-        } else if (abbrMap.has(matched)) {
-          const node = inlineAbbrMatches(matched, txt, abbrMap);
-          df.appendChild(node);
-        } else {
-          // FAIL -- not sure that this can really happen
-          pub(
-            "error",
-            `Found token '${matched}' but it does not correspond to anything`
-          );
-        }
+    let matched = true;
+    for (const t of subtxt) {
+      matched = !matched;
+      if (!matched) {
+        df.appendChild(document.createTextNode(t));
+      } else if (t.startsWith("{{{")) {
+        const node = inlineXrefMatches(t);
+        df.appendChild(node);
+      } else if (t.startsWith("[[")) {
+        const nodes = inlineBibrefMatches(t, txt, conf);
+        df.append(...nodes);
+      } else if (abbrMap.has(t)) {
+        const node = inlineAbbrMatches(t, txt, abbrMap);
+        df.appendChild(node);
+      } else if (
+        /MUST(?:\s+NOT)?|SHOULD(?:\s+NOT)?|SHALL(?:\s+NOT)?|MAY|(?:NOT\s+)?REQUIRED|(?:NOT\s+)?RECOMMENDED|OPTIONAL/.test(
+          t
+        )
+      ) {
+        const node = inlineRFC2119Matches(t);
+        df.appendChild(node);
+      } else {
+        // FAIL -- not sure that this can really happen
+        throw `Found token '${t}' but it does not correspond to anything`;
```

Comment:
Always throw real errors, otherwise we get not stack! :) 

```suggestion
        throw new Error(`Found token '${t}' but it does not correspond to anything`);
```

Reply:
Thanks, I will read about the difference between simple `throw` and creating a new error.